### PR TITLE
[PRIM][PIR] Support all vjp gen

### PR DESCRIPTION
**Pull request by Charles-hit**
### PR types

New features
### PR changes

Others 
### Description

Pcard-66975
support all vjp gen

### CODE REVIEW

File: paddle/fluid/operators/generator/type_mapping.py [hunk 1]
```diff
@@ -86,8 +86,8 @@
 }
 
 optional_output_type_map = {
-    'Tensor': 'const paddle::optional<Tensor>&',
-    'Tensor[]': 'const paddle::optional<std::vector<Tensor>>&',
+    'Tensor': 'const paddle::optional<Tensor>',
+    'Tensor[]': 'const paddle::optional<std::vector<Tensor>>',
```
Comment (Aurelius84) [inline paddle/fluid/operators/generator/type_mapping.py:90]: 这里为什么把引用去掉了，但const 还在？

**Review by Aurelius84 — APPROVED** [review]
Great Work!

**Review by heavyrain-lzy — APPROVED** [review]
LGTM

File: paddle/fluid/operators/generator/type_mapping.py (hunk 1 above)
Comment (Charles-hit) [reply paddle/fluid/operators/generator/type_mapping.py:90]: 感谢提醒，为了不阻塞大家工作，下个pr修改一下

**Pull request merged**